lib/msal-browser/src/cache/BrowserStorage.ts

```diff
@@ -109,80 +126,42 @@ export class BrowserStorage extends CacheManager {
      * @param value
      */
     setItem(key: string, value: string | object, type: string): void {
-        // save the cacheItem
-        switch (type) {
-            case CacheSchemaType.ACCOUNT:
-            case CacheSchemaType.CREDENTIAL:
-            case CacheSchemaType.APP_META_DATA:
-                this.windowStorage.setItem(key, JSON.stringify(value));
-                break;
-            case CacheSchemaType.TEMPORARY: {
-                const stringVal = value as string;
-                this.windowStorage.setItem(key, stringVal);
-                if (this.cacheConfig.storeAuthStateInCookie) {
-                    this.setItemCookie(key, stringVal);
-                }
-                break;
-            }
-            case CacheSchemaType.TELEMETRY: {
-                this.windowStorage.setItem(key, JSON.stringify(value));
-                break;
-            }
-            default: {
-                throw BrowserAuthError.createInvalidCacheTypeError();
+
+        let stringifiedValue: string;
+        // throw for invalid cache type
+        if (!(type in CacheSchemaType) || (type === CacheSchemaType.UNDEFINED)) {
+            throw BrowserAuthError.createInvalidCacheTypeError();
+        }
+
+        // store cookies if TEMPORARY Cache, else stringify and store
+        if (type === CacheSchemaType.TEMPORARY) {
+            stringifiedValue = value as string;
+            if(this.cacheConfig.storeAuthStateInCookie) {
+                this.setItemCookie(key, stringifiedValue);
             }
+        } else {
+            stringifiedValue = JSON.stringify(value);
         }
+
+        // actual window storage
+        this.windowStorage.setItem(key, stringifiedValue);
     }
 
+    // getItem ->
```

Comment:
If this comment is meant as documentation, can we make it a block comment? I assume the linter will complain about multi-lined comments.

Reply:
temporary: will delete